pkg/node/address.go

```diff
@@ -357,6 +358,19 @@ func SetIPv4AllocRange(net *cidr.CIDR) {
 	})
 }
 
+func GetIPAMAllocPools() []ipamTypes.IPAMPoolAllocation {
```

Comment:
I realise that's entirely non-obvious, but these package global functions are historic, and we should avoid adding new ones. They are here for compatability reasons, while we transition the code-base to a more modular approach. See https://github.com/cilium/cilium/pull/21191 for a bit of history.

Instead, depend on the local node store and call update on it, if you need to set something.

Reply:
Let me look into the approach you are suggesting. I was following the changes from [this](https://github.com/cilium/cilium/pull/27285/files#diff-6cb81d56efa78ad58fc0dd7038341c90aad7382fe7fd160282f6fec7ab014ca3R26-R31) recent PR.

Reply:
Commit 0329660 follows the recommendations of https://github.com/cilium/cilium/pull/27100#discussion_r1307355084 by using `LocalCiliumNodeResource` to trigger events instead of the internal `Node` type.